### PR TITLE
Expressing releases using weak until

### PR DESCRIPTION
**Pull request by gian21391**
I added the method create_weak_until (and create_and for readability purposes) in order to express the release operator using the weak until:

(a)R(b) = (b)W((a)&(b))

### CODE REVIEW

File: include/copycat/ltl.hpp [hunk 1]
```diff
@@ -454,6 +432,20 @@ class ltl_formula_store
...
 
+  ltl_formula create_weak_until( ltl_formula const& a, ltl_formula const& b )
+  {
+    /* (a)W(b) = ((a)U(b))|G(a) */
+    auto until = create_until(a, b);
+    auto globally = create_globally(a);
+    return create_or(until, globally);
+  }
+
+  ltl_formula create_releases( ltl_formula const& a, ltl_formula const& b )
```
Comment (hriener) [inline include/copycat/ltl.hpp]: Should be named `releases`.

File: include/copycat/ltl.hpp [hunk 2]
```diff
@@ -376,33 +381,6 @@ class ltl_formula_store
     return {index,0};
   }
 
-  ltl_formula create_releases( ltl_formula const& a, ltl_formula const& b )
```
Comment (hriener) [inline include/copycat/ltl.hpp]: Should not be removed.

File: include/copycat/ltl.hpp [hunk 3]
```diff
@@ -316,6 +316,11 @@ class ltl_formula_store
     return {index,0};
   }
 
+  ltl_formula create_and( ltl_formula const& a, ltl_formula const& b )
+  {
+      return !create_or(!a, !b);
```
Comment (hriener) [inline include/copycat/ltl.hpp]: Missing space, i.e., `!create_or( !a, !b )`.

File: include/copycat/ltl.hpp [hunk 4]
```diff
@@ -454,6 +432,20 @@ class ltl_formula_store
     return !eventually( !a );
   }
 
+  ltl_formula create_weak_until( ltl_formula const& a, ltl_formula const& b )
+  {
+    /* (a)W(b) = ((a)U(b))|G(a) */
+    auto until = create_until(a, b);
+    auto globally = create_globally(a);
+    return create_or(until, globally);
```
Comment (hriener) [inline include/copycat/ltl.hpp]: Missing spaces (as before).
`until` and `globally` could be `const`.

File: include/copycat/ltl.hpp [hunk 5]
```diff
@@ -454,6 +432,20 @@ class ltl_formula_store
     return !eventually( !a );
   }
 
+  ltl_formula create_weak_until( ltl_formula const& a, ltl_formula const& b )
```
Comment (hriener) [inline include/copycat/ltl.hpp]: Should be named `weak_until`.

**Comment by gian21391** [comment]
I created a new class `ltl_operators` which includes the methods `make_and`, `eventually`, `globally`, `weak_until`, `releases`. Additionally `releases` now implements (a)R(b) = !((!a)U(!b))

**Comment by hriener** [comment]
This looks good. However, your contribution breaks the test. I will fix that after merging.

**Comment by hriener** [comment]
Thanks!

**Pull request merged**